chain/x/zoracle/internal/keeper/querier_test.go

```diff
@@ -140,6 +142,123 @@ func TestQueryRequestIncompleteValidator(t *testing.T) {
 	require.Equal(t, acs, acsBytes)
 }
 
+func TestQueryDataSourceById(t *testing.T) {
+	ctx, keeper := CreateTestInput(t, false)
+	keeper.SetMaxDataSourceExecutableSize(ctx, 20)
+	// Create variable "querier" which is a function
+	querier := NewQuerier(keeper)
+
+	// query before add a data source
+	_, err := querier(
+		ctx,
+		[]string{"data_source", "1"},
+		abci.RequestQuery{},
+	)
+	// Should return error data source not found
+	require.NotNil(t, err)
+
+	owner := sdk.AccAddress([]byte("owner"))
+	name := "data_source"
+	fee := sdk.NewCoins(sdk.NewInt64Coin("uband", 10))
+	executable := []byte("executable")
+	expectedResult := types.NewDataSourceQuerierInfo(1, owner, name, fee, executable)
+
+	// Add a new data source
+	err = keeper.AddDataSource(ctx, expectedResult.Owner, expectedResult.Name, expectedResult.Fee, expectedResult.Executable)
+	require.Nil(t, err)
+
+	// This time querier should be able to find a data source
+	dataSource, err := querier(
+		ctx,
+		[]string{"data_source", "1"},
+		abci.RequestQuery{},
+	)
+	require.Nil(t, err)
+
+	expectedResultBytes, errJSON := codec.MarshalJSONIndent(keeper.cdc, expectedResult)
+	require.Nil(t, errJSON)
+
+	require.Equal(t, expectedResultBytes, dataSource)
+}
+
+func TestQueryDataSourcesByStartIdAndNumberOfDataSources(t *testing.T) {
+	ctx, keeper := CreateTestInput(t, false)
+	keeper.SetMaxDataSourceExecutableSize(ctx, 20)
+	// Create variable "querier" which is a function
+	querier := NewQuerier(keeper)
+
+	expectedResult := []types.DataSourceQuerierInfo{}
+
+	// Add a new 10 data sources
+	for i := 1; i <= 10; i++ {
+		owner := sdk.AccAddress([]byte("owner" + strconv.Itoa(i)))
+		name := "data_source_" + strconv.Itoa(i)
+		fee := sdk.NewCoins(sdk.NewInt64Coin("uband", 10))
+		executable := []byte("executable" + strconv.Itoa(i))
+		eachDataSource := types.NewDataSourceQuerierInfo(int64(i), owner, name, fee, executable)
+
+		err := keeper.AddDataSource(ctx, eachDataSource.Owner, eachDataSource.Name, eachDataSource.Fee, eachDataSource.Executable)
```

Comment:
ditto

Reply:
Why is this not address? @taobun @prin-r

Reply:
Leave it to be `AddDataSource` because of this function increase data source count, which makes check all data sources count condition work in get all data sources function.